extensions/objects/models/objects.py

```diff
@@ -839,7 +839,13 @@ class Units(BaseObject):
 
     SCHEMA = {
         'type': 'dict',
-        'properties': []
+        'properties': [{
```

Comment:
Don't modify this schema. Keep it as a simple dict.

This conflicts with NumberWithUnitsObjectFactory's schema. So I suggest that you store unit not as a unit object factory object but use the value of the unit key in the dict returned by the toDict() method.